addons/board/static/tests/dashboard_tests.js

```diff
@@ -1039,4 +1041,107 @@ QUnit.test('click on a cell of pivot view inside dashboard', async function (ass
+                    fieldDescription: "Date",
+                    rangeId: "this_month",
+                    rangeDescription: "This Month",
+                    range: ["&",["date", ">=", "2020-07-01"], ["date", "<", "2020-08-01"]],
+                    comparisonRangeId: "previous_period",
+                    comparisonRange: ["&", ["date", ">=", "2020-06-01"], ["date", "<", "2020-07-01"]],
+                    comparisonRangeDescription: "Previous Period",
+                });
+                return Promise.resolve(true);
+            }
+            return this._super.apply(this, arguments);
+        },
+    });
+
+    await actionManager.doAction({
+        id: 1,
+        res_model: 'partner',
+        type: 'ir.actions.act_window',
+        views: [[false, 'pivot']],
+    });
+
+    // compare this month and previous month
+    await cpHelpers.toggleTimeRangeMenu(actionManager);
+    await cpHelpers.toggleTimeRangeMenuBox(actionManager);
+    await cpHelpers.applyTimeRange(actionManager);
+
+    // add the view to the dashboard
+    await cpHelpers.toggleFavoriteMenu(actionManager);
+
+    await testUtils.dom.click($('.o_add_to_board > button'));
+    await testUtils.fields.editInput($('.o_add_to_board input'), 'a name');
+    await testUtils.dom.click($('.o_add_to_board div button'));
+
+    unpatchDate();
+    actionManager.destroy();
+});
+
+QUnit.test('correctly display the time range descriptions of a reporting view in comparison mode', async function (assert) {
+    assert.expect(1);
+
+    this.data.partner.fields.date = { string: 'Date', type: 'date', sortable: true };
+    this.data.partner.records[0].date = '2020-07-15';
+
+    const form = await createView({
+        View: BoardView,
+        model: 'board',
+        data: this.data,
+        arch: `<form string="My Dashboard">
+                <board style="2-1">
+                    <column>
+                        <action string="ABC" name="51"></action>
+                    </column>
+                </board>
+            </form>`,
+        archs: {
+            'partner,1,pivot':
+                '<pivot string="Partner"></pivot>',
+        },
+        mockRPC: function (route, args) {
+            if (route === '/board/static/src/img/layout_1-1-1.png') {
+                return Promise.resolve();
+            }
+            if (route === '/web/action/load') {
+                return Promise.resolve({
+                    context: '{"time_ranges":{"fieldName":"date","fieldDescription":"Date",' +
+                            '"rangeId":"this_month","rangeDescription":"This Month",' +
+                            '"range":["&",["date",">=","2020-07-01"],["date","<","2020-08-01"]],' +
+                            '"comparisonRangeId":"previous_period",' +
+                            '"comparisonRange":["&",["date",">=","2020-06-01"],["date","<","2020-07-01"]],' +
+                            '"comparisonRangeDescription":"Previous Period"}}',
```

Comment:
suggestion: JSON.stringify({...}) to improve readability

Reply:
done